nncf/experimental/onnx/graph/nncf_graph_builder.py

```diff
@@ -0,0 +1,136 @@
+
+from nncf.experimental.onnx.graph.onnx_graph import ONNXGraph
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import ONNX_OPERATION_METATYPES
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import ConstantMetatype
+
+
+class GraphConverter:
+    """
+    Builds the NNCFGraph from an ONNX model
+    """
+
+    @staticmethod
+    def create_nncf_graph(onnx_model: ModelProto) -> NNCFGraph:
+        """
+        Adds all ONNX nodes from 'onnx_model' and then adds thr special input_nodes and output_nodes.
+        """
+
+        nncf_graph = NNCFGraph()
+        onnx_graph = ONNXGraph(onnx_model)
+        for node in onnx_graph.get_all_nodes():
+            node_name = node.name
+            node_type = node.op_type
+            metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(node_type)
+            if metatype == ConstantMetatype:  # We don't need to quantize Constants
+                continue
+            nncf_graph.add_nncf_node(node_name=node_name,
+                                     node_type=node_type,
+                                     node_metatype=metatype,
+                                     layer_attributes=None)
+        input_counter = {}
+        output_counter = {}
+        for output_node in nncf_graph.get_all_nodes():
+            output_node_id = output_node.node_id
+            outputs = onnx_graph.get_all_node_outputs(output_node.node_name)
+            for output in outputs:
+                nodes = onnx_graph.get_nodes_by_input(output)
+                shape = onnx_graph.get_node_output_shape(output)
+                for in_node in nodes:
+                    in_node_id = nncf_graph.get_node_by_name(in_node.name).node_id
+                    input_counter[in_node_id] = input_counter.get(in_node_id, -1) + 1
+                    output_counter[output_node_id] = input_counter.get(output_node_id, -1) + 1
+                    nncf_graph.add_edge_between_nncf_nodes(
+                        from_node_id=output_node_id,
+                        to_node_id=in_node_id,
+                        tensor_shape=shape,
+                        input_port_id=input_counter[in_node_id],
+                        output_port_id=output_counter[output_node_id],
+                        dtype=Dtype.FLOAT
+                    )
+        # Add Input Nodes
+        for i, _input in enumerate(onnx_graph.get_all_model_inputs()):
+            tensor_type = _input.type.tensor_type
+            input_shape = []
+            if tensor_type.HasField("shape"):
+                for d in tensor_type.shape.dim:
+                    if d.HasField("dim_value"):
+                        input_shape.append(int(d.dim_value))
+                    else:
+                        raise RuntimeError('There is no integer value in model Input dimension')
+            else:
+                raise RuntimeError('There is no shape in model Inputs')
+            input_node = nncf_graph.add_nncf_node(node_name='input_node_' + str(i),
```

Comment:
Please follow the naming from the other frameworks: `nncf_input_node_N`, `nncf_output_node_N`

Reply:
Done